src/rdiffbackup/actions/__init__.py

```diff
@@ -558,33 +561,42 @@ def _operate_regress(self, try_regress=True, noticeable=False):
         if Globals.get_api_version() < 201:  # compat200
             if self.repo.needs_regress_compat200():
                 if not try_regress:
-                    return 1
+                    return Globals.RET_CODE_ERR
                 log.Log("Previous backup seems to have failed, regressing "
                         "destination now", log.WARNING)
                 try:
                     self.repo.base_dir.conn.regress.Regress(self.repo.base_dir)
-                    return 0
+                    return Globals.RET_CODE_OK
                 except Security.Violation:
                     log.Log(
                         "Security violation while attempting to regress "
                         "destination, perhaps due to --restrict-read-only or "
                         "--restrict-update-only", log.ERROR)
-                    return 1
+                    return Globals.RET_CODE_ERR
             else:
                 log.Log("Given repository doesn't need to be regressed",
                         regress_verbosity)
-                return 0  # all is good
+                return Globals.RET_CODE_OK
         else:
             if self.repo.needs_regress():
                 if not try_regress:
-                    return 1
+                    return Globals.RET_CODE_ERR
                 log.Log("Previous backup seems to have failed, regressing "
                         "destination now", log.WARNING)
                 return self.repo.regress()
+            elif force:
+                if self.repo.force_regress():
```

Comment:
That function already existed ?

Reply:
No, you missed the crunchy part:smile:

Added line 299 of repository.py